File: Arbitration/MPArbitration/Model/Holiday.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    public class Holiday
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [MaxLength(50)]
        [JsonPropertyName("country")]
        public string Country { get; set; } = string.Empty;
        
        [JsonPropertyName("endDate")]
        public DateTime? EndDate { get; set; } = null;

        [MaxLength(100)]
        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [MaxLength(50)]
        [JsonPropertyName("region")]
        public string Region { get; set; } = string.Empty;

        [JsonPropertyName("startDate")]
        public DateTime? StartDate { get; set; } = null;

    }
}

File: Arbitration/MPArbitration/ClientApp/src/app/model/holiday.ts
export class Holiday {
    public id = 0;

    public country = '';
    public endDate: Date|undefined;
    public name = '';
    public region ='';
    public startDate: Date|undefined;

    constructor(obj?:any) {
        if(!obj)
            return;
        Object.assign(this, JSON.parse(JSON.stringify(obj)));
        if(this.endDate)
            this.endDate = new Date(obj.endDate); 
        if(this.startDate)
            this.startDate = new Date(obj.startDate);
    }
}